src/layout/rectangles.js

```diff
@@ -33,10 +32,15 @@ export default function(layoutStrategy) {
         }
 
         selection.each(function(data, index) {
+
+            var g = dataJoin(this, data);
+            g.call(component);
+
             // obtain the rectangular bounding boxes for each child
-            var childRects = data.map(function(d, i) {
-                var childPos = position(d, i);
-                var childSize = size(d, i);
+            var childRects = g[0].map(function(node, i) {
```

Comment:
That's a good point, I was meaning to re-visit this. I can't think of a better way to map a d3 selection to an array. Have you seen any alternatives?